client/packages/host/src/components/Footer/UserDetails.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { FC } from 'react';
+import {
+  Box,
+  CircularProgress,
+  FlatButton,
+  PaperPopoverSection,
+  useAuthContext,
+  usePaperClickPopover,
+  useTranslation,
+  useNavigate,
+  useUserDetails,
+  useConfirmationModal,
+  RouteBuilder,
+  PowerIcon,
+  TextWithLabelRow,
+} from '@openmsupply-client/common';
+import { AppRoute } from '@openmsupply-client/config';
+import { PropsWithChildrenOnly } from '@common/types';
+
+export const AdminSelector: FC<PropsWithChildrenOnly> = ({ children }) => {
+  const { logout, user, token } = useAuthContext();
+  const navigate = useNavigate();
+  const { hide, PaperClickPopover } = usePaperClickPopover();
+  const { isLoading } = useUserDetails(token);
+  const t = useTranslation();
+
+  const handleLogout = () => {
+    navigate(RouteBuilder.create(AppRoute.Login).build());
+    logout();
+  };
+
+  const showConfirmation = useConfirmationModal({
+    onConfirm: handleLogout,
+    message: t('messages.logout-confirm'),
+    title: t('heading.logout-confirm'),
+  });
+
+  const logoutButton = (
+    <FlatButton
+      startIcon={<PowerIcon fontSize="small" color="primary" />}
+      label={t('logout')}
+      disabled={false}
+      onClick={async () => {
+        hide();
+        showConfirmation();
+      }}
+      sx={{
+        whiteSpace: 'nowrap',
+        overflowX: 'hidden',
+        overflowY: 'visible',
+        textOverflow: 'ellipsis',
+      }}
+    />
+  );
+
+  return user ? (
+    <PaperClickPopover
+      placement="top"
+      width={350}
+      Content={
+        <PaperPopoverSection label={`${user.firstName} ${user.lastName}`}>
+          {isLoading ? (
+            <CircularProgress size={12} />
+          ) : (
+            <Box
+              style={{
+                overflowY: 'auto',
+                maxHeight: 300,
+                gap: 10,
+                margin: '.5rem',
+              }}
+            >
+              <TextWithLabelRow
+                label={t('heading.username')}
+                text={user.name}
+                textProps={{ textAlign: 'left', lineHeight: 1.5 }}
+                labelProps={{
+                  sx: { textAlign: 'left', width: 80, lineHeight: 1.5 },
+                }}
+              />
+              <TextWithLabelRow
+                label={t('label.email')}
+                text={user.email ?? ''}
+                textProps={{ textAlign: 'left', lineHeight: 1.5 }}
+                labelProps={{
+                  sx: { textAlign: 'left', width: 50, lineHeight: 1.5 },
+                }}
+              />
```

Comment:
Regarding comment: `I had to overwrite some styles for textprops and labelprops to make label and text align. The email now wraps funny if the text is too long. Open to suggestions.`

Love that we're trying to solve this! We can try converting this into an [ellipsis](https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow) with CSS. Just not sure if that's a pattern that we follow @lache-melvin. 

But if we humour the idea of solving it that way. We can utilise the sx prop within `TextWithLabelRow` (not the one on lableProps). And apply the necessary css fields.

Reply:
After looking further with the props. I wonder if we can just pass in `textOverflow: ellipsis` to textProps with a width

Reply:
Yes we do! Here's an example - important to add the Tooltip if we add the ellipsis so the can see the full email if they need 😁 

https://github.com/msupply-foundation/open-msupply/blob/aeb2842227dd456a9b1f4eaf7288bee648e48cbd/client/packages/common/src/ui/layout/tables/components/Cells/TooltipTextCell/TooltipTextCell.tsx#L11-L15

Reply:
Thanks @GeronimoJohn and @lache-melvin for your help on this! I've decided to leave the tooltip placement as right as it looks the cleanest.